src/common.rs

```diff
@@ -304,3 +306,52 @@ impl Frame<'static> {
         usize::from(self.width) * usize::from(self.height)
     }
 }
+
+enum Quantizer {
+    // Use NeuQuant for high-colour images...
+    NQ(color_quant::NeuQuant),
+    // And an exact look-up for <= 256 colors.
+    Exact {
+        // Palette look-up table.
+        lookup: HashMap<(u8, u8, u8, u8), usize>,
+        // A palette that can be used by Frame.
+        palette: Vec<u8>,
+    },
+}
+
+impl Quantizer {
+    fn index_of(&self, pixel: &[u8]) -> usize {
+        match self {
+            Quantizer::NQ(nq) => nq.index_of(pixel),
+            Quantizer::Exact { lookup, palette: _ } =>
```

Comment:
You can use `Quantizer::Exact { lookup, .. }` to skip the irrelevant fields

Reply:
Thanks, fixed.